src/Command/GenerateEnvironmentCommand.php

```diff
@@ -284,90 +309,135 @@ function ($environment_url) {
+        $input->setOption('dump-always-update', $always_update_ref_dump);
 
         try {
-          $ssh_config_name = $input->getOption('ssh-config-name');
+          $always_update_ref_dump = $input->getOption('dump-always-update') ? (bool) $input->getOption('dump-always-update') : FALSE;
         } catch (\Exception $error) {
           $this->getIo()->error($error->getMessage());
 
           return 1;
         }
 
-        if (!$ssh_config_name) {
-          $ssh_config_name = $this->getIo()->ask(
-            'What is the name for the dump remote server in your ~/.ssh/config file?',
-            array_key_exists('COMBAWA_DB_FETCH_CNX_STRING', $envVars) ? $envVars['COMBAWA_DB_FETCH_CNX_STRING'] : 'my_remote'
+        $retrieval_tool = $this->getIo()->choice(
+          'When updated, what is the tool used to retrieve the reference dump?',
+          ['cp', 'scp'],
+          array_key_exists('COMBAWA_DB_RETRIEVAL_TOOL', $envVars) ? $envVars['COMBAWA_DB_RETRIEVAL_TOOL'] : 'scp'
+        );
+        $input->setOption('dump-retrieval-tool', $retrieval_tool);
+
+        if ($retrieval_tool == 'scp') {
+
+          $use_ssh_config_name = $this->getIo()->confirm(
+            'Do you have an SSH config name from your ~/.ssh/config to use to retrieve the dump?',
+            FALSE
           );
-          $input->setOption('ssh-config-name', $ssh_config_name);
+
+          if ($use_ssh_config_name) {
+            try {
+              $ssh_config_name = $input->getOption('ssh-config-name');
+            } catch (\Exception $error) {
+              $this->getIo()->error($error->getMessage());
+
+              return 1;
+            }
+
+            if (!$ssh_config_name) {
+              $ssh_config_name = $this->getIo()->ask(
+                'What is the name of you config entry in your ~/.ssh/config file?',
+                array_key_exists('COMBAWA_DB_FETCH_SSH_CONFIG_NAME', $envVars) ? $envVars['COMBAWA_DB_FETCH_SSH_CONFIG_NAME'] : 'my_remote'
+              );
+              $input->setOption('ssh-config-name', $ssh_config_name);
+            }
+          }
+          else {
+            try {
+              $ssh_connection_info = $input->getOption('scp-connection-info');
+            } catch (\Exception $error) {
+              $this->getIo()->error($error->getMessage());
+
+              return 1;
+            }
+
+            if (!$ssh_connection_info) {
+              $ssh_connection_info = $this->getIo()->ask(
+                'What is the connection string to the remote server?',
+                array_key_exists('COMBAWA_DB_FETCH_SCP_CONNECTION', $envVars) ? $envVars['COMBAWA_DB_FETCH_SCP_CONNECTION'] : 'user@server.org'
+              );
+              $input->setOption('scp-connection-info', $ssh_connection_info);
+            }
+          }
         }
 
+        $validateDumpExtension = function ($path) {
+          switch (pathinfo($path, PATHINFO_EXTENSION)) {
+            case 'gz':
+              return TRUE;
```

Comment:
This should return the input value. Currently, this returns `TRUE` that's converted to `1` in the final output.